bin/create-release

```diff
@@ -0,0 +1,197 @@
+
+
+################################################################################
+# Resolve and validate GitHub token, check auth
+################################################################################
+
+with open("gh-release-token.txt", "r") as file:
+    GH_TOKEN = file.read().rstrip()
+
+if not GH_TOKEN.startswith("ghp_"):
+    print("Invalid token found in gh-release-token.txt", file=sys.stderr)
+    exit(1)
+
+check_gh_auth_response = requests.get(
+    "https://api.github.com/repos/seattle-uat/civiform",
+    headers={"Authorization": "token " + GH_TOKEN})
+if check_gh_auth_response.status_code != 200:
+    print(
+        "GitHub API auth failed, satus code: " +
+        str(check_gh_auth_response.status_code),
+        file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Check docker hub auth
+################################################################################
+
+if not "Login Succeeded" in shell_cmd(
+        "docker login") or not "civiform" in shell_cmd("docker info"):
+    print("Unable to login to docker hub")
+    exit(1)
+
+################################################################################
+# Set release variables
+################################################################################
+
+COMMIT_SHA = sys.argv[1]
+RELEASE_NUMBER = sys.argv[2]
+RELEASER_NAME = shell_cmd("git config --get user.name")
+RELEASER_EMAIL = shell_cmd("git config --get user.email")
+
+################################################################################
+# Validate release version number
+################################################################################
+
+if not re.match("^v\d+\.\d+\.\d+$", RELEASE_NUMBER):
+    print("Invalid version number: " + RELEASE_NUMBER, file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Validate commit SHA
+################################################################################
+
+branches_containing_commit_sha = shell_cmd(
+    "git --no-pager branch --contains " + COMMIT_SHA)
+branches_containing_commit_sha = [
+    re.sub('\*', '', x).strip()
+    for x in branches_containing_commit_sha.split("\n")
+]
+if not "main" in branches_containing_commit_sha:
+    print("Invalid commit SHA: " + COMMIT_SHA, file=sys.stderr)
+    exit(1)
+
+################################################################################
+# Find snapshot tag that matches commit SHA
+################################################################################
+
+# Git defaults to using the first 7 chars of the full SHA for the short SHA.
+SHORT_SHA = shell_cmd("git rev-parse --short " + COMMIT_SHA)[:7]
+
+print(f"Searching for an image tag with short SHA {SHORT_SHA}")
+
+tags_url = "https://registry.hub.docker.com/v2/repositories/civiform/civiform/tags?order=-last_activity&page_size=100"
+snapshot_tag = None
```

Comment:
Nit, not blocking: Re-initialize snapshot_tag as the first line in the loop. This will prevent weird bugs where we go through the loop a second time and for some reason use a previous snapshot_tag version.

That may not actually be a problem here, but generally I prefer this type of pattern.

Reply:
done